app/assets/javascripts/watchlist.js

```diff
@@ -0,0 +1,431 @@
+
+        window.open(`mailto: ${new_instances[user_id]["email"]}`, "_blank");
+        console.log(new_instances[user_id]["instance_ids"]);
+        update_watchlist(method, new_instances[user_id]["instance_ids"]);
+      })
+
+      $('.ui.button.resolve_single').click(function() {
+        method = "resolve";
+        var user_id = $(this).parent().parent().attr('id');
+        var instances = get_active_instances(new_instances, contacted_instances);
+        update_watchlist(method, instances[user_id]["instance_ids"]);
+      })
+  });
+
+  $('#contact_button').click(function(){
+    method = "contact";
+
+    var emails = [];
+    selected_user_ids.forEach(user_id => {
+      emails.push(new_instances[user_id]["email"]);
+    });
+
+    var instance_ids = [];
+    selected_user_ids.forEach(user_id => {
+      instance_ids = instance_ids.concat(new_instances[user_id]["instance_ids"]);
+    });
+  
+    if (instance_ids.length > 0) {
+      window.open(`mailto: ${emails}`, "_blank");
+      update_watchlist(method, instance_ids);
+    }
+  });
+
+  $('#resolve_button').click(function(){
+    method = "resolve";
+    var instances = get_active_instances(new_instances, contacted_instances);
+
+    var instance_ids = [];
+    selected_user_ids.forEach(user_id => {
+      instance_ids = instance_ids.concat(instances[user_id]["instance_ids"]);
+    });
+
+    if (instance_ids.length > 0) {
+      update_watchlist(method, instance_ids);
+    }
+  });
+}
+
+// Uncheck all checkboxes when moving to another tab
+$('.ui.vertical.fluid.tabular.menu .item').on('click', function() {
+  $('.ui.tab.segments.active .ui.checkbox').checkbox('uncheck');
+  $('.ui.checkbox.select_all').checkbox('uncheck');
+  $('.ui.vertical.fluid.tabular.menu .item').removeClass('active');
+  $(this).addClass('active');
+
+  switch ($(this).attr("data-tab")) {
+    case "new_tab":
+      $("#contact_button").removeClass("disabled");
+      $("#resolve_button").removeClass("disabled");
+      break;
+    case "contacted_tab":
+      $("#contact_button").addClass("disabled");
+      $("#resolve_button").removeClass("disabled");
+      break;
+    case "resolved_tab":
+      $("#contact_button").addClass("disabled");
+      $("#resolve_button").addClass("disabled");
+      break;
+    case "archived_tab":
+      $("#contact_button").addClass("disabled");
+      $("#resolve_button").addClass("disabled");
+      break;
+    default:
+      console.log(`${$(this).attr("data-tab")} is not a valid tab`);
+      return;
+  }
+});
+
+function get_active_instances(new_instances, contacted_instances) {
+  var tab = $(".ui.tab.segments.active").attr('id');
+  
+  switch(tab){
+    case "new_tab":
+      return new_instances;
+    case "contacted_tab":
+      return contacted_instances;
+    default:
+      console.log(`${tab} is not a valid tab for this action`)
+      return;
+  }
+}
+
+// TODO: update logic here when contacting / resolving students
```

Comment:
I wasn't too sure what to update with this function. It's not being used right now (as far as I know), so I was thinking about just removing it. If someone sees a purpose for it though, please let me know!

Reply:
I think this was a boiler plate by @amartinez116.  Seeing that you have written your own update_watchlist it should be fine?

Reply:
ah yes, I wrote this here to help your PR out in updating students in the watchlist when the instructor contacts/resolves them.

Reply:
oh wait after taking a closer look: this was the old version of the update_watchlist function. The correct version is the one finalized at the bottom. Line 393. Please use that one and make sure to delete lines 374-391.

Reply:
Got it, thanks for letting me know! I removed it